### PR TITLE
Update typescript version

### PR DESCRIPTION
**Pull request by sharafian**
Closes https://github.com/interledgerjs/ilp-connector/issues/480

This fixes errors when importing ILP Connector to the latest version of typescript. Also makes the type checking stricter by enabling `noImplicitAny`

This doesn't break anything if you're still importing from typescript 2.6

### CODE REVIEW

File: src/stores/leveldown.ts [hunk 1]
```diff
@@ -23,9 +23,14 @@ class LeveldownStore implements StoreInstance {
     this.db = levelup(leveldown(path))
   }
 
-  async get (key: string) {
+  async get (key: string): Promise<string | undefined> {
```
Comment (sentientwaffle) [inline src/stores/leveldown.ts]: Why does this have `| undefined`?

File: src/services/admin-api.ts [hunk 2]
```diff
@@ -153,7 +153,7 @@ export default class AdminApi {
     return balanceMiddleware.getStatus()
   }
 
-  private async postBalance (url: string, _data: object) {
+  private async postBalance (url: string | undefined, _data: object) {
```
Comment (sentientwaffle) [inline src/services/admin-api.ts]: Why `| undefined`?

Comment (sharafian) [reply src/services/admin-api.ts]: `req.url` is `string | undefined` now, so the url params need to become `string | undefined`

File: src/stores/leveldown.ts (hunk 1 above)
Comment (sharafian) [reply src/stores/leveldown.ts]: if the key doesn't exist then it returns `undefined`

**Pull request merged**